api/app/signals/apps/questionnaires/rest_framework/views/public.py

```diff
@@ -119,3 +121,13 @@ def get_object(self):
             raise Gone('Expired!')
 
         return session
+
+    @action(detail=True, url_path=r'submit/?$', methods=['POST', ])
+    def submit(self, request, *args, **kwargs):
+        # TODO: calls to this endpoint are not idempotent, investigate whether
+        # they should be.
```

Comment:
I think this call should only work for Sessions that ar not `frozen` yet.
A 404 can be raised if the Session does not exists and perhaps the `Gone` could be raised if the Session is frozen?